lib/ecmascript.ts

```diff
@@ -1326,7 +1326,9 @@ export const ES = ObjectAssign({}, ES2020, {
     if (offsetOpt === 'reject') {
       const offsetStr = ES.FormatTimeZoneOffsetString(offsetNs);
       const timeZoneString = ES.IsTemporalTimeZone(timeZone) ? GetSlot(timeZone, TIMEZONE_ID) : 'time zone';
-      throw new RangeError(`Offset ${offsetStr} is invalid for ${dt} in ${timeZoneString}`);
+      // TODO: The tsc emit for this line rewrites to invoke the PlainDateTime's valueOf method, NOT
```

Comment:
`throw new RangeError(`Offset ${offsetStr} is invalid for ${dt} in ${timeZoneString}`);` is transpiled by tsc to `throw new RangeError('Offset ' + offsetStr + ' is invalid for ' + dt + ' in ' + timeZoneString);`, and the '+' operator invokes `valueOf`.

This would be an issue for either language given that the problem occurs regardless of who is using tsc to compile the sources - us when we bundle the sources for packaging or others I suppose if they directly compile against our TS source.

Is having `valueOf` throwing something officially part of the Temporal spec, and thus something that the native object will do as well?

Reply:
> Is having `valueOf` throwing something officially part of the Temporal spec, and thus something that the native object will do as well?

Yes. https://tc39.es/proposal-temporal/#sec-temporal.plaindate.prototype.valueof. That's why I'm concerned about this.

I'll add this issue to the agenda for the next Temporal Champions meeting. We may want to keep the behavior as-is, but transpiling string interpolation to `+` might be popular enough (perhaps not only in TS, maybe in Babel too for regular JS?) that we may want to revisit.

Reply:
To clarify, this isn't an issue for this PR-- it's a more general issue for Temporal overall. See https://github.com/tc39/proposal-temporal/issues/1681. Feel free to resolve this conversation.

Reply:
Thanks for opening that issue. I've added a reference to the TypeScript repo issue so if someone stumbles across this fix in the future and finds out it's been fixed upstream we can revert this fix.